src/NBitcoin/ChainedHeader.cs

```diff
@@ -4,6 +4,63 @@
 
 namespace NBitcoin
 {
+    /// <summary>
+    /// Represents the state of the availability of a block.
+    /// </summary>
+    public enum BlockDataAvailabilityState
+    {
+        /// <summary>
+        /// A <see cref="BlockHeader"/> is presented, the block data is not available.
+        /// </summary>
+        HeaderOnly,
+
+        /// <summary>
+        /// We are interested in downloading the <see cref="Block"/> that is being represented by the current <see cref="BlockHeader"/>. 
+        /// This happens when a header is a part of the chain that can potentially replace our consensus tip 
```

Comment:
i can't of like "can potentially replace", but i see you are aiming for little more correct wording, i agree with that as well

therefore i'd try

`This happens when we don't have block which is represented by this header and the header is a part of a chain that can potentially replace our consensus tip because its chain work is greater than our consensus tip's chain work.`